### PR TITLE
feat: add TypeScript definition

### LINKED ISSUE #55: TypeScript support
From #39 

TypeScript support will be useful for users.

Probably, We will add separated `.d.ts` (TypeScript Ambient Declarations) to express-mongo-sanitize.

- [sindresorhus/typescript-definition-style-guide: Style guide for adding type definitions to my npm packages](https://github.com/sindresorhus/typescript-definition-style-guide)

### PR DESCRIPTION
**Pull request by azu**
Add `index.d.ts` and add testing using [tsd](https://github.com/SamVerschueren/tsd#expecterrorfunction)

fix #55

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,9 +3,15 @@
   "version": "2.0.2",
   "description": "Sanitize your express payload to prevent MongoDB operator injection.",
   "main": "index.js",
+  "types": "index.d.ts",
+  "files": [
```
Comment (azu) [inline package.json:7]: the published package will only include `index.js` and `index.d.ts`.
So, we do not need to includes `index.test-d.ts` and `test.js`.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * Create middleware instance
+     * @param options
+     */
+    (options?: ExpressMongoSanitize.Options): Handler;
+    /**
+     * Remove any keys containing prohibited characters
+     * @param target
+     * @param options
+     */
+    sanitize<T extends object>(target: T, options?: ExpressMongoSanitize.Options): T;
```
Comment (fiznool) [inline index.d.ts:23]: Should we use something other than object here? Maybe Record<string, unknown>? Or { [key: string]: unknown } ?

Also see: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Check if the payload has keys with prohibited characters‘
+     * @param target
+     */
+    has(target: object): boolean;
```
Comment (fiznool) [inline index.d.ts:28]: As above?

**Review by fiznool — APPROVED** [review]
Just a couple of smaller comments otherwise lgtm

File: index.d.ts (hunk 2 above)
Comment (azu) [reply index.d.ts]: Thanks.
Change to `Record<string, unknown> | unknown[]`.

**Comment by azu** [comment]
@fiznool Can you publish it?
Or, please add [me](https://www.npmjs.com/~azu) as Collaborators on https://www.npmjs.com/package/express-mongo-sanitize

**Pull request merged**